src/components/YourFeed.tsx

```diff
@@ -164,11 +214,18 @@ export class YourFeed extends React.PureComponent<DispatchProps & StateProps, Yo
         this.props.onRefreshPosts();
     }
 
-    private async unfollowFeed(author: Author) {
-        const confirmUnfollow = await AreYouSureDialog.show('Are you sure you want to unfollow?');
-        if (confirmUnfollow) {
-            this.props.onUnfollowFeed(author.uri);
-            this.props.navigation.goBack(null);
+    private onFollowPressed = async (author: Author) => {
+        const followedFeed = this.props.feeds.find(feed => feed.feedUrl === author.uri);
+        if (followedFeed != null) {
```

Comment:
I don't really understand what's going on here: sometimes if the follow pressed we unfollow a feed, sometimes we add the feed to the feeds list.
In the unfollow case I think the naming is confusing and maybe the UI as well.
In the other knownFeed case, it seems we add the knownFeed to the feed list one more time, because the only case when the `onAddFeed` does anything is in the `FeedContainer`, where `knownFeeds` is just a filtered list of `state.feeds`.